Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/returntypes/RCFComputeDescriptor.java

```diff
@@ -53,15 +54,15 @@ public class RCFComputeDescriptor extends Point {
     // the actual dimensions
     int dimension;
 
-    // the inputlength; useful for standalone analysis
+    // the input length; useful for standalone analysis
     int inputLength;
 
     // sequence index (the number of updates to RCF) -- it is possible in imputation
     // that
     // the number of updates more than the input tuples seen by the overall program
     long totalUpdates;
 
-    // determines if values can be inputed and or expected point calculated
+    // determines if values can be input and or expected point calculated
     boolean reasonableForecast;
```

Comment:
Can you explain more about this field? When do you use reasonableForecast?

Reply:
Great q :) We do not use it in any more ThresholdedRCF (TRCF) -- it was used in the prior versions when the forecast was more rudimentary. It does make sense that "we can say it's not right but we cannot quite say what the correct course is" -- but I think that nuance can be too much. Note that the predict() function now provides a clustered nearest neighbors and can be useful in answering the nuanced question. In TRCF, if the dimension of the space is 4 then we can forecast reasonably. Also when we have seen enough data for anomaly scoring (outputAfter) and transformations being stable -- then perhaps this field is not useful. I did not what to remove the field to break backward compatibility (but happy to). We should also discuss this restriction of 4. I did feel strongly that the calling function should be cognizant of the quality of forecast -- but it it true that such a decision can create a lot more work for the calling function (and a simpler solution is to advice users to not have 1 dimension and shingle size less than 4, of something).